sdk/lib/drivers/ip/network/ports.c

```diff
@@ -39,6 +39,12 @@ VOID DeallocatePort( PPORT_SET PortSet, ULONG Port ) {
     ASSERT(Port >= PortSet->StartingPort);
     ASSERT(Port < PortSet->StartingPort + PortSet->PortsToOversee);
```

Comment:
Why assert when deallocating but not when allocating?

Reply:
Isn't it because the TCPAllocatePort function already ensures that? While here the assert is part of checking whether the caller of the DeallocatePort gives a valid Port value?

Reply:
> ... TCPAllocatePort ... DeallocatePort ...

I don't think your "questions" explain why alloc and dealloc (should continue to) behave differently with regard to input validation(s).